week7/cooper_client_react/01_cooper_calculator.md

```diff
@@ -0,0 +1,543 @@
+
+We will store our component tests in a separate folder (called `__tests__`) and use a different set of testing frameworks. 
+
+Let's go through the setup by creating folders and adding dependecies. Execute the following commands in your terminal:
+
+```shell
+$ npm i -D enzyme enzyme-adapter-react-16 react-test-renderer
+$ touch src/setupTests.js
+$ mkdir src/__tests__
+$ touch src/__tests__/displayCooperResult.test.js
+```
+
+Add this to the files we created above:
+
+```js
+// setupTests.js
+import { configure } from 'enzyme';
+import Adapter from 'enzyme-adapter-react-16';
+
+configure({ adapter: new Adapter() }); 
+```
+
+```js
+// displayCooperResult.test.js
+import React from 'react';
+import { shallow } from 'enzyme';
+
+import DisplayCooperResult from '../Components/DisplayCooperResult';
+
+describe('<DisplatCooperResult />', () => {
+  it('evaluates the correct result for female/poor', () => {
+    const describedComponent = shallow(<DisplayCooperResult distance="1000" gender="female" age="23"/>);
+    const response = <p>Result: Poor</p>
+    expect(describedComponent.contains(response)).toEqual(true)
+  })
+
+  it('evaluates the correct result for female/average', () => {
+    const describedComponent = shallow(<DisplayCooperResult distance="2000" gender="female" age="23"/>);
+    const response = <p>Result: Average</p>
+    expect(describedComponent.contains(response)).toEqual(true)
+  })
+})
+```
+
+After doing the BMI Calculator we know how to send in props to a child component. We send in `distance`, `gender` and `age`. Then we expect to get a result that the component renders.
+
+If you run the component tests now (using the `npm run test` command in the terminal), you will see that the test complains that it can't find the `DisplayCooperResult` module. Let's create it:
+
+`$ mkdir src/Components`
+
+`$ touch src/Components/DisplayCooperResult.js`
+
+To start with, we will add a stateful/class component with a `render` method. We will expand it with more functionality as we move ahead.
+
+The component should look like this:
+
+```js
+import React, { Component } from 'react';
+
+class DisplayCooperResult extends Component {
+
+  render() {
+    return (
+      <div>
+      </div>
+    )
+  }
+}
+
+export default DisplayCooperResult
+```
+
+When you run the component tests now you should get another error. The new error states that our expectation/assertion is returning `false`. The component is not returning what it is supposed to when it gets the values from the input fields. It is now time to add some logic to our `DisplayCooperResult` component. We will store that logic in a separate file in a folder we will create for that very purpose. Please look carefully at the imports below, to figure out where that is. 
```

Comment:
Break this paragraph up for better readability, perhaps here :
It is now time to add some logic to our `DisplayCooperResult`